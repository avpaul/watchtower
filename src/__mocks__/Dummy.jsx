
export default [
  {
    id: 'gfgdvfuhdjhfhgdf',
    type: 'App\\Notifications\\FellowStatus',
    notifiable_id: 5,
    data: {
      status: 'onTrack',
      level: 'D0B',
      message: 'Hi Ronald, you are on Track for this week',
      email: 'ronald.mutebi@andela.com',
    },
    read_at: null,
    created_at: '2018-11-01 14:34:03',
    updated_at: '2018-10-29 14:34:03',
  },
  {
    id: 'rtrgfgdvfuhdjhfhgdf',
    type: 'App\\Notifications\\FellowStatus',
    notifiable_id: 5,
    data: {
      status: 'oFFTrack',
      level: 'D0B',
      message: 'Hi Bilango, you are off Track for this week',
      email: 'bilango.fake@andela.com',
    },
    read_at: null,
    created_at: '2018-10-31 14:34:03',
    updated_at: '2018-10-31 14:34:03',
  },
  {
    id: 'fsrtrgfgdvfuhdjhfhgdf',
    type: 'App\\Notifications\\FellowStatus',
    notifiable_id: 5,
    data: {
      status: 'oFFTrack',
      level: 'D0B',
      message: 'Hi mac, you are off Track for this week',
      email: 'mac.pc@andela.com',
    },
    read_at: null,
    created_at: '2018-10-29 14:34:03',
    updated_at: '2018-10-29 14:34:03',
  },
  {
    id: 'gfgdvfuhdjhfhgdf',
    type: 'App\\Notifications\\FellowStatus',
    notifiable_id: 5,
    data: {
      status: 'onTrack',
      level: 'D0B',
      message: 'Hi Ronald, you are on Track for this week',
      email: 'ronald.mutebi@andela.com',
    },
    read_at: null,
    created_at: '2018-10-27 14:34:03',
    updated_at: '2018-10-29 14:34:03',
  },
  {
    id: 'gfgdvfuhdjhfhgdf',
    type: 'App\\Notifications\\FellowStatus',
    notifiable_id: 5,
    data: {
      status: 'onTrack',
      level: 'D0B',
      message: 'Hi Ronald, you are on Track for this week',
      email: 'ronald.mutebi@andela.com',
    },
    read_at: null,
    created_at: '2018-10-29 14:34:03',
    updated_at: '2018-10-29 14:34:03',
  },
];
