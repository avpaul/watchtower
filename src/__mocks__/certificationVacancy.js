export const getCertificationVacancies = (count = 1) => {
  const vacancies = [];
  for (let i = 0; i < count; i += 1) {
    vacancies.push({
      certification: {
        id: i + 1,
        name: 'Mr. Freddie Doyle',
        description: 'Aut eaque voluptate ut eum autem autem quis.',
        exclusive: false,
        duration: 20,
        deleted_at: null
      },
      vacancy_details: {
        id: 1,
        certification_id: 1,
        is_active: false,
        created_at: '2019-08-24 22:51:15',
        updated_at: '2019-08-24 23:08:35',
        requester: 'collins.muru@andela.com',
        start_date: '2019-08-31 00:00:00',
        end_date: '2019-09-14 00:00:00',
        closing_date: '2019-08-23 00:00:00',
        fellow_id: null,
        cycle_id: i + 1,
        deleted_at: null,
        applications: []
      },
      available_slots: 2
    });
  }
  return vacancies;
};

export default getCertificationVacancies;
