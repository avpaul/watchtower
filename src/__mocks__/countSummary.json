{
  "D0A Simulations": {
    "Week 1": {
      "onPip": 0,
      "onTrack": 33,
      "offTrack": 12
    },
    "Week 2": {
      "onPip": 0,
      "onTrack": 29,
      "offTrack": 16
    },
    "Week 3": {
      "onPip": 0,
      "onTrack": 40,
      "offTrack": 5
    },
    "Week 4": {
      "onPip": 0,
      "onTrack": 38,
      "offTrack": 7
    },
    "Week 5": {
      "onPip": 0,
      "onTrack": 38,
      "offTrack": 7
    },
    "Week 6": {
      "onPip": 0,
      "onTrack": 35,
      "offTrack": 10
    },
    "Week 7": {
      "onPip": 0,
      "onTrack": 34,
      "offTrack": 11
    },
    "Week 8": {
      "onPip": 0,
      "onTrack": 31,
      "offTrack": 14
    }
  },
  "D0B Apprenticeship": {
    "Week 1": {
      "onPip": 0,
      "onTrack": 31,
      "offTrack": 9
    },
    "Week 2": {
      "onPip": 0,
      "onTrack": 28,
      "offTrack": 12
    },
    "Week 3": {
      "onPip": 0,
      "onTrack": 27,
      "offTrack": 13
    },
    "Week 4": {
      "onPip": 0,
      "onTrack": 26,
      "offTrack": 9
    },
    "Week 5": {
      "onPip": 0,
      "onTrack": 19,
      "offTrack": 11
    },
    "Week 6": {
      "onPip": 0,
      "onTrack": 16,
      "offTrack": 9
    },
    "Week 7": {
      "onPip": 0,
      "onTrack": 15,
      "offTrack": 5
    },
    "Week 8": {
      "onPip": 0,
      "onTrack": 10,
      "offTrack": 5
    },
    "Week 9": {
      "onPip": 0,
      "onTrack": 13,
      "offTrack": 2
    },
    "Week 10": {
      "onPip": 0,
      "onTrack": 10,
      "offTrack": 0
    },
    "Week 11": {
      "onPip": 0,
      "onTrack": 5,
      "offTrack": 0
    },
    "Week 12": {
      "onPip": 0,
      "onTrack": 1,
      "offTrack": 4
    }
  }
}
