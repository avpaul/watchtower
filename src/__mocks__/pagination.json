{
  "perPage": 10,
  "page": 1,
  "pages": 3,
  "firstPageURL": "http://localhost:8000/api/fellows?page=1",
  "nextPageURL": "http://localhost:8000/api/fellows?page=2",
  "prevPageURL": "http://localhost:8000/api/fellows?page=1",
  "finalPageURL": "http://localhost:8000/api/fellows?page=3",
  "currentPage": 1
}
