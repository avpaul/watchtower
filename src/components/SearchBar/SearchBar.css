.fellow-table-search {
  height: 45px;
  max-width: 209px;
  border-radius: 4px;
  margin-bottom: 24px;
}

/* enable absolute positioning */
.table_search_div {
  margin-left: unset;
  margin-right: 25px;
}

.table_search_div > p {
  font-size: 14px;
  font-family: 'DIN Pro Light';
}

.table-search-append {
  width: 45px;
  background-color: #ffaf30;
  opacity: 0.62;
  cursor: default;
}
.result-count {
  margin-left: 60px;
  font-family: 'DIN Pro Light', Arial, Helvetica, sans-serif;
  font-size: 1.25em;
  line-height: 1.2em;
  font-weight: 400;
  margin-bottom: 27px;
}

.result-count span.border-bottom {
  border-color: #3359db !important;
  border-width: 2px !important;
}

.clear-filters {
  color: #0459e4;
  font-size: 16px;
  line-height: 21px;
}
/* Mobile and Tablet screen */
@media screen and (max-width: 767px) {
  .fellow-table-search {
    width: 200px;
  }
}
