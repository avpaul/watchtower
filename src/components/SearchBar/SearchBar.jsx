import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { DebounceInput } from 'react-debounce-input';
import './SearchBar.css';

class SearchBar extends Component {
  componentDidUpdate(prevProps) {
    const { getFellows, perPage, filter, search } = this.props;
    if (prevProps.search !== search) {
      getFellows({ perPage, filter, search });
    }
  }

  handleClick = event => {
    const { search, handleSearchChange } = this.props;
    event.preventDefault();
    handleSearchChange({ target: { value: search } });
  };

  renderSearchButton = () => (
    <span className="input-group-append">
      <button
        id="search-button"
        className="btn border-0 table-search-append"
        onClick={this.handleClick}
        type="button"
      >
        <i className="fa fa-search text-white " />
      </button>
    </span>
  );

  renderSearchInput = () => {
    const { search, handleSearchChange } = this.props;

    return (
      <div id="table_search_form" className="table_search_div">
        <p>Search</p>
        <div className="input-group fellow-table-search">
          <DebounceInput
            minLength={2}
            debounceTimeout={300}
            type="search"
            className="form-control py-2 h-100 border-0"
            name="fellow_search"
            value={search}
            onChange={handleSearchChange}
            placeholder="Search the Table"
          />
          {this.renderSearchButton()}
        </div>
      </div>
    );
  };

  render() {
    return <Fragment>{this.renderSearchInput()}</Fragment>;
  }
}

SearchBar.propTypes = {
  getFellows: PropTypes.func.isRequired,
  handleSearchChange: PropTypes.func.isRequired,
  perPage: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  filter: PropTypes.shape().isRequired,
  search: PropTypes.string.isRequired
};

export default SearchBar;
