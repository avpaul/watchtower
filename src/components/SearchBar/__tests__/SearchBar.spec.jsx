import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import SearchBar from '../SearchBar';
import { perPage, page } from '../../../__mocks__/pagination';
import {
  ONTRACK,
  OFFTRACK_WK4_MINUS
} from '../../../redux/constants/fellowFilters';

const props = {
  search: '',
  perPage,
  page,
  results: 15,
  filter: ONTRACK,
  getFellows: jest.fn(),
  handleSearchChange: jest.fn()
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchBar {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the search bar', () => {
  const wrapper = shallow(<SearchBar {...props} />);
  expect(wrapper).toMatchSnapshot();
});
it('makes the API call when the search term changes', () => {
  const getFellowsSpy = jest.fn();
  const wrapper = shallow(<SearchBar {...props} getFellows={getFellowsSpy} />);
  wrapper.setProps({ filter: OFFTRACK_WK4_MINUS });
  expect(getFellowsSpy).not.toHaveBeenCalled();
  const callArgs = { perPage, filter: OFFTRACK_WK4_MINUS, search: 'foo' };
  wrapper.setProps({ search: 'foo' });
  expect(getFellowsSpy).toHaveBeenCalledWith(callArgs);
});

it('calls handleClick when the search button is clicked', () => {
  const event = { preventDefault: jest.fn };
  const wrapper = shallow(<SearchBar {...props} search="search" />);
  const handleClickSpy = jest.spyOn(wrapper.instance(), 'handleClick');
  wrapper.instance().handleClick(event);
  expect(handleClickSpy).toHaveBeenCalledWith(event);
});
