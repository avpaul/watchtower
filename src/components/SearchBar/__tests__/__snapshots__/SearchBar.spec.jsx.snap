// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the search bar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    filter="onTrack"
    getFellows={[Function]}
    handleSearchChange={[Function]}
    page={1}
    perPage={10}
    results={15}
    search=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="table_search_div"
        id="table_search_form"
>
        <p>
                Search
        </p>
        <div
                className="input-group fellow-table-search"
        >
                <DebounceInput
                        className="form-control py-2 h-100 border-0"
                        debounceTimeout={300}
                        element="input"
                        forceNotifyByEnter={true}
                        forceNotifyOnBlur={true}
                        inputRef={undefined}
                        minLength={2}
                        name="fellow_search"
                        onBlur={undefined}
                        onChange={[Function]}
                        onKeyDown={undefined}
                        placeholder="Search The Table"
                        type="search"
                        value=""
                />
                <span
                        className="input-group-append"
                >
                        <button
                                className="btn border-0 table-search-append"
                                id="search-button"
                                onClick={[Function]}
                                type="button"
                        >
                                <i
                                        className="fa fa-search text-white "
                                />
                        </button>
                </span>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Search
</p>,
          <div
            className="input-group fellow-table-search"
>
            <DebounceInput
                        className="form-control py-2 h-100 border-0"
                        debounceTimeout={300}
                        element="input"
                        forceNotifyByEnter={true}
                        forceNotifyOnBlur={true}
                        inputRef={undefined}
                        minLength={2}
                        name="fellow_search"
                        onBlur={undefined}
                        onChange={[Function]}
                        onKeyDown={undefined}
                        placeholder="Search The Table"
                        type="search"
                        value=""
            />
            <span
                        className="input-group-append"
            >
                        <button
                                    className="btn border-0 table-search-append"
                                    id="search-button"
                                    onClick={[Function]}
                                    type="button"
                        >
                                    <i
                                                className="fa fa-search text-white "
                                    />
                        </button>
            </span>
</div>,
        ],
        "className": "table_search_div",
        "id": "table_search_form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
          },
          "ref": null,
          "rendered": "Search",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <DebounceInput
                className="form-control py-2 h-100 border-0"
                debounceTimeout={300}
                element="input"
                forceNotifyByEnter={true}
                forceNotifyOnBlur={true}
                inputRef={undefined}
                minLength={2}
                name="fellow_search"
                onBlur={undefined}
                onChange={[Function]}
                onKeyDown={undefined}
                placeholder="Search The Table"
                type="search"
                value=""
/>,
              <span
                className="input-group-append"
>
                <button
                                className="btn border-0 table-search-append"
                                id="search-button"
                                onClick={[Function]}
                                type="button"
                >
                                <i
                                                className="fa fa-search text-white "
                                />
                </button>
</span>,
            ],
            "className": "input-group fellow-table-search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "form-control py-2 h-100 border-0",
                "debounceTimeout": 300,
                "element": "input",
                "forceNotifyByEnter": true,
                "forceNotifyOnBlur": true,
                "inputRef": undefined,
                "minLength": 2,
                "name": "fellow_search",
                "onBlur": undefined,
                "onChange": [Function],
                "onKeyDown": undefined,
                "placeholder": "Search The Table",
                "type": "search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn border-0 table-search-append"
                  id="search-button"
                  onClick={[Function]}
                  type="button"
>
                  <i
                                    className="fa fa-search text-white "
                  />
</button>,
                "className": "input-group-append",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fa fa-search text-white "
/>,
                  "className": "btn border-0 table-search-append",
                  "id": "search-button",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-search text-white ",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "button",
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="table_search_div"
          id="table_search_form"
>
          <p>
                    Search
          </p>
          <div
                    className="input-group fellow-table-search"
          >
                    <DebounceInput
                              className="form-control py-2 h-100 border-0"
                              debounceTimeout={300}
                              element="input"
                              forceNotifyByEnter={true}
                              forceNotifyOnBlur={true}
                              inputRef={undefined}
                              minLength={2}
                              name="fellow_search"
                              onBlur={undefined}
                              onChange={[Function]}
                              onKeyDown={undefined}
                              placeholder="Search The Table"
                              type="search"
                              value=""
                    />
                    <span
                              className="input-group-append"
                    >
                              <button
                                        className="btn border-0 table-search-append"
                                        id="search-button"
                                        onClick={[Function]}
                                        type="button"
                              >
                                        <i
                                                  className="fa fa-search text-white "
                                        />
                              </button>
                    </span>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Search
</p>,
            <div
              className="input-group fellow-table-search"
>
              <DebounceInput
                            className="form-control py-2 h-100 border-0"
                            debounceTimeout={300}
                            element="input"
                            forceNotifyByEnter={true}
                            forceNotifyOnBlur={true}
                            inputRef={undefined}
                            minLength={2}
                            name="fellow_search"
                            onBlur={undefined}
                            onChange={[Function]}
                            onKeyDown={undefined}
                            placeholder="Search The Table"
                            type="search"
                            value=""
              />
              <span
                            className="input-group-append"
              >
                            <button
                                          className="btn border-0 table-search-append"
                                          id="search-button"
                                          onClick={[Function]}
                                          type="button"
                            >
                                          <i
                                                        className="fa fa-search text-white "
                                          />
                            </button>
              </span>
</div>,
          ],
          "className": "table_search_div",
          "id": "table_search_form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
            },
            "ref": null,
            "rendered": "Search",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <DebounceInput
                  className="form-control py-2 h-100 border-0"
                  debounceTimeout={300}
                  element="input"
                  forceNotifyByEnter={true}
                  forceNotifyOnBlur={true}
                  inputRef={undefined}
                  minLength={2}
                  name="fellow_search"
                  onBlur={undefined}
                  onChange={[Function]}
                  onKeyDown={undefined}
                  placeholder="Search The Table"
                  type="search"
                  value=""
/>,
                <span
                  className="input-group-append"
>
                  <button
                                    className="btn border-0 table-search-append"
                                    id="search-button"
                                    onClick={[Function]}
                                    type="button"
                  >
                                    <i
                                                      className="fa fa-search text-white "
                                    />
                  </button>
</span>,
              ],
              "className": "input-group fellow-table-search",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "form-control py-2 h-100 border-0",
                  "debounceTimeout": 300,
                  "element": "input",
                  "forceNotifyByEnter": true,
                  "forceNotifyOnBlur": true,
                  "inputRef": undefined,
                  "minLength": 2,
                  "name": "fellow_search",
                  "onBlur": undefined,
                  "onChange": [Function],
                  "onKeyDown": undefined,
                  "placeholder": "Search The Table",
                  "type": "search",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn border-0 table-search-append"
                    id="search-button"
                    onClick={[Function]}
                    type="button"
>
                    <i
                                        className="fa fa-search text-white "
                    />
</button>,
                  "className": "input-group-append",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="fa fa-search text-white "
/>,
                    "className": "btn border-0 table-search-append",
                    "id": "search-button",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-search text-white ",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "button",
                },
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
